src/webui/webui.c

```diff
@@ -1597,6 +1597,271 @@ page_play_auth(http_connection_t *hc, const char *remain, void *opaque)
+  return deviceid;
+}
+
+
+/**
+ * @param fail_log_reason Log this reason if permissions fail.
+ * @return Permission for the verified user (caller owns the memory) or NULL.
+ */
+__attribute__((warn_unused_result))
+static access_t *hdhomerun_verify_user_permission(const http_connection_t *hc,
+                                                  const char *fail_log_reason)
+{
+  /* Not explicitly enabled?  Then all calls fail. */
+  if (!config.hdhomerun_server_enable) {
+    tvhwarn(LS_WEBUI, "hdhomerun server not enabled but received request [%s]",
+            fail_log_reason?:"");
+    return NULL;
+  }
+
+  const char *hdhr_user = config.hdhomerun_server_username ?: "";
+  access_t *perm = access_get(hc->hc_peer, hdhr_user, hdhomerun_server_verify_callback, NULL);
+
+  if (access_verify2(perm, ACCESS_STREAMING)) {
+    /* Failed */
+    tvhwarn(LS_WEBUI, "hdhomerun server received request but no streaming permission for user [%s] [%d] [%s]",
+            hdhr_user ?: "<none>",
+            perm? perm->aa_rights : 0,
+            fail_log_reason?:"");
+    access_destroy(perm);
+    return NULL;
+  } else {
+    return perm;
+  }
+}
+
+
+/**
+ * Return the discovery information for HDHomeRun to give clients
+ * details of how to access the lineup.
+ *
+ * Our HDHomeRun server implementation uses a server configured
+ * username to determine access rather than passing it through the
+ * request.  This is because HDHomeRun clients are usually configured
+ * with only an IP address and port number and do not offer any input
+ * for credentials.
+ */
+static int
+hdhomerun_server_discover(http_connection_t *hc, const char *remain, void *opaque)
+{
+  access_t *perm = hdhomerun_verify_user_permission(hc, "discover");
+  if (!perm)
+    return http_noaccess_code(hc);
+
+  char http_ip[128];
+  htsbuf_queue_t *hq = &hc->hc_reply;
+  const char *server_name = hdhomerun_get_server_name();
+  const uint32_t deviceid = hdhomerun_get_deviceid();
+
+  tcp_get_str_from_ip(hc->hc_self, http_ip, sizeof(http_ip));
+
+  /* The contents below for the discovery message are based on tvhProxy */
+  htsbuf_append_str(hq, "{ ");
+  htsbuf_append_and_escape_jsonstr_nv(hq, "FriendlyName", server_name);
+  htsbuf_qprintf(hq, ", "                                               \
+                 "\"BaseURL\" : \"http://%s:%u\", "                     \
+                 "\"DeviceAuth\": \"3xw5UaJXhVShHEBoy76FuYQi\", "       \
+                 "\"DeviceID\": \"%08X\", "                             \
+                 "\"FirmwareName\": \"hdhomerun_atsc\", "               \
+                 "\"FirmwareVersion\": \"20200101\", "                  \
+                 "\"LineupURL\": \"http://%s:%u/lineup.json\", "        \
+                 "\"Manufacturer\": \"tvheadend\", "                    \
+                 "\"ModelNumber\": \"HDTC-2US\", "                      \
+                 "\"TunerCount\": 6 "                                   \
```

Comment:
Unfortunately it is not possible to identify the tuner count. For the SAT>IP Server there are fields to enter the number of advertised tuners but I don't think the number actually matters for a client.

Reply:
At least i know plex counts how many are in use so i guess it will stop when it reaches that number. But will tvheadend work if i change that number? I did and did configure and autobuild.sh and the installed and after that i could not tune to any channels, be it in plex or directly in tvheadend. Might be the build so i could try again...

Reply:
@Flole998 Oww, I thought TVheadend keeps track of the free tuners. In any case I know for certain that at least Plex does take the tuner count into account when connecting to HDHomeRun. Having it show more tuners than are available may also produce funny errors on the clients connected to it (Although I'm unaware of how HDHomeRun signals when when the tuners are in use from another application). I would suggest to add a menu entry for setting this similar to the "SAT>IP Server" then, if not now, at least in the future.

Reply:
Keeping track of tuners works until you use IPTV, then you have basically an infinite amount of tuners. I don't think the infinity sign would be accepted by plex though ;)

Does Plex properly distinguish between Tuners and channels? You can watch 10 Channels on 1 Tuner in certain cases.

Reply:
IPTV has a pissibility to set limits based on your supplyer when you add it to the network. And tvheadend counts and if you reach the limit you set it wont tune to that channel. It works good in tvheadend but dont know how it works in combination whith plex setting a limit aswell.

Reply:
It's a possibility, not a requirement. So if someone didn't set it the tuner count is infinite, which is obviously not a valid value. That's why automatic counting of tuners is not an option.

Reply:
I see.